### DISCUSSION TITLE
Add `data-testId` prop to elements for testing purposes

### DISCUSSION THREAD

**Discussion by annette-graham** [primefaces/community#1094] [PrimeReact] [answered]
I have been struggling to properly write unit tests both the `FileUpload` and `DataTable` components. I think if elements (e.g the inputs) had a `data-testId` prop it would be much easier to test using React Testing Library. This way I can hopefully test uploading files, editing table data etc without running into the errors I currently am.

If not applied by default, then maybe providing an option to pass our own test ids to the components would suffice?

If there are easy ways to test in-depth functionality of the named components, please point me in the direction of how as I am currently very much at a loss!

Thank you :)

**Comment by melloware** [comment, marked as answer]
A lot of components already allow Passthrough to set the data-testid like: https://github.com/primefaces/primereact/issues/4676

**Reply by annette-graham** [reply]
Amazing, thank you!
